src/smefit/optimize/ns.py

```diff
@@ -289,5 +297,14 @@ def save(self, result):
             for c in self.coefficients.name:
                 values[c].append(self.coefficients[c].value)
 
-        with open(self.results_path / "posterior.json", "w", encoding="utf-8") as f:
+        posterior_file = self.results_path / "posterior.json"
+
+        # if it s a single parameter fit check if the posterior file is already present and in case it is update it
```

Comment:
this code is a bit repeated can you move it to the same place both for NS and MC?